src/metabase/driver/h2.clj

```diff
@@ -397,6 +400,30 @@
   [driver [_ field]]
   [:log10 (sql.qp/->honeysql driver field)])
 
+(defn- literal-text-value?
+  [[_ value {base-type :base_type effective-type :effective_type} :as clause]]
```

Comment:
Is it possible for `effective-type` to be present here? I was basing this off [the implementation of `->honeysql [:sql :value]`](https://github.com/metabase/metabase/blob/expression-literals-fix-h2/src/metabase/driver/sql/query_processor.clj#L587) which similarly destructures a `:value` clause, but I recently realized that the [legacy schema for `:value`](https://github.com/metabase/metabase/blob/expression-literals-fix-h2/src/metabase/legacy_mbql/schema.cljc#L191) does not include `:effective_type` as a valid option.

Reply:
Your guess is as good as mine. I think drivers tend to operate on `:database_type` and `:base_type`. `:semantic_type` and `:effective_type` are more the QP's business.

I gather the idea is that the QP understands high-level `:effective_type`s, eg. that we want this date which is stored as a string to get turned into a date so we can compare it to another date. The QP *inserts cast expressions* accordingly, leaving the driver only to deal (relatively locally) with the details of how to make that conversion for that specific driver.